### ISSUE TITLE
Structural re-arrangement for section 3

### ISSUE THREAD

**Issue by jimsch** [sacmwg/draft-ietf-sacm-architecture#47]
I would like to finish the re-arrangment of section 3 that was started in Prague and never finished.  To this end I would propose doing the following:
1.  Move section 3.1.3 (providers and consumers) to after section 3.1.4 (controls)
2.  Move broker, proxy and repository into the new section 3.2
3.  Rename the section to be "Example Components"

This makes it more uniform in terms of discussion the specifics of roles and then generalities of Components as a general concept.  Among other this I would no longer have a reason to argue with Lisa about issues such as "Is a Proxy really a controller" because it has aspects of all three component rolls

If you are interested in this, I can put together a pull request on which details can be commented on.

**Comment by ncamwing** [comment]
Yes, this looks like a good direction.  Thanks for putting the this together here so we can review!
